lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -6,21 +6,20 @@ class PodsProjectGenerator
         # relative support files.
         #
         class PodTargetInstaller < TargetInstaller
-          # @return [Hash{Pathname => Pathname}] A hash of all umbrella headers, grouped by the directory
-          #         the are stored in. This can be `nil` if there is no grouping.
+          # @return [Pathname] The directory in which umbrella headers are stored
           #
-          attr_reader :umbrella_headers_by_dir
+          attr_reader :umbrella_headers_dir
 
           # Initialize a new instance
           #
           # @param [Sandbox] sandbox @see TargetInstaller#sandbox
           # @param [Pod::Project] project @see TargetInstaller#project
           # @param [Target] target @see TargetInstaller#target
-          # @param [Hash{Pathname => Pathname}] umbrella_headers_by_dir @see #umbrella_headers_by_dir
+          # @param [Pathname] umbrella_headers_dir @see #umbrella_headers_dir
           #
-          def initialize(sandbox, project, target, umbrella_headers_by_dir = nil)
+          def initialize(sandbox, project, target, umbrella_headers_dir = nil)
```

Comment:
This was accepting a hash of all the header directories, where it really just needs the directory for this target

Reply:
it was a hash of paths, _keyed_ by directory